fxrecorder/src/lib/perfherder.rs

```diff
@@ -0,0 +1,46 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use crate::analysis::VisualMetrics;
+use serde_json::{json, Value};
+
+/// Generate a JSON blob containing the performance metrics for Perfherder.
+pub fn generate_perfherder_metrics(metrics: &VisualMetrics) -> Value {
+    json!({
+      "application": {
+        "name": "Firefox",
```

Comment:
The capitalization matters here so this needs to be `firefox`: https://searchfox.org/mozilla-central/source/testing/mozharness/external_tools/performance-artifact-schema.json#8